hs_core/models.py

```diff
@@ -2675,30 +2676,37 @@ def resource(self):
         """Return content_object representing the resource from a resource file."""
         return self.content_object
 
+    def size(self):
+        if self._size == 0:
+            self.calculate_size
+        return self._size
+
     # TODO: write unit test
     @property
-    def size(self):
+    def calculate_size(self):
         """Return file size for federated or non-federated files."""
         if self.resource.resource_federation_path:
             if __debug__:
                 assert self.resource_file.name is None or \
                     self.resource_file.name == ''
             try:
-                return self.fed_resource_file.size
+                self._size = self.fed_resource_file.size
             except SessionException:
                 logger = logging.getLogger(__name__)
                 logger.warn("file {} not found".format(self.storage_path))
-                return 0
+                self._size = 0
         else:
             if __debug__:
                 assert self.fed_resource_file.name is None or \
                     self.fed_resource_file.name == ''
             try:
-                return self.resource_file.size
+                self._size = self.resource_file.size
             except SessionException:
                 logger = logging.getLogger(__name__)
                 logger.warn("file {} not found".format(self.storage_path))
-                return 0
+                self._size = 0
+        self.save()
+        return self._size
```

Comment:
Probably don't need this return.

Reply:
I think it's fair to leave it in.  If someone ends up calling this method, they'll probably want the size returned.

Reply:
@sblack-usu Should a method is more appropriate than a property given the name ```calculate_size()```?

Reply:
yes, you're right.  Now that I'm making another change, I'll go ahead and remove the return.